projects/roscoe/baselines/scores.py

```diff
@@ -0,0 +1,347 @@
+
+
+######### BartScore (and its variants)
+# Note: You might have to load some of its dependencies yourself.
+# HuggingFace is the major one
+# Second argument here should be path to `bart_score.py` of the BARTScore repo
+try:
+    bart_score = SourceFileLoader("bart_score", BART_SCORE_REPO).load_module()
+    from bart_score import BARTScorer
+except ImportError:
+    raise ImportError(f"Run `bart-score not found. Make sure it's in {BART_SCORE_REPO}")
+
+
+class BartscoreBase(BaselineScorer):
+    def __init__(self):
+        raise NotImplementedError(
+            'Must init with a "self.scorer" that loads the right model and a "self.score_type"'
+        )
+
+    def get_scores(self, score_me):
+        scores = self.scorer.score(srcs=score_me.hypos, tgts=score_me.context_ref)
+        return {self.score_type: scores}
+
+
+@register_scorer([BARTSCORE_F])
+class BartscoreBaselineScorer(BartscoreBase):
+    def __init__(self):
+        self.scorer = BARTScorer(
+            device=DEFAULT_DEVICE, checkpoint='facebook/bart-large'
+        )
+        self.score_type = BARTSCORE_F
+
+
+@register_scorer([BARTSCORE_CNN_F])
+class BartscoreBaselineCNNScorer(BartscoreBase):
+    def __init__(self):
+        self.scorer = BARTScorer(
+            device=DEFAULT_DEVICE, checkpoint='facebook/bart-large-cnn'
+        )
+        self.score_type = BARTSCORE_CNN_F
+
+
+@register_scorer([BARTSCORE_CNN_PARA_F])
+class BartscoreCNNParaBaselineScorer(BartscoreBase):
+    def __init__(self):
+        self.scorer = BARTScorer(
+            device=DEFAULT_DEVICE, checkpoint='facebook/bart-large-cnn'
+        )
+        # Path here should be to fine tuned BART model from https://github.com/neulab/BARTScore#direct-use
+        self.scorer.load("/private/home/mpchen/BARTScore/bart_score_para_finetuned.pth")
+        self.score_type = BARTSCORE_CNN_PARA_F
+
+
+@register_scorer([BARTSCORE_FINETUNED_F])
+class BartscoreFineTunedBaselineScorer(BartscoreBase):
+    def __init__(self):
+        """
+        NOTE: In the bart_score.py file above, BARTScorer.load() needs to be
+        ```
+        def load(self, path=None):
+            if path is None:
+                path = 'models/bart.pth'
+            print("Loading from finetuned with custom hacks", path)
+            state_dict = torch.load(path, map_location=DEFAULT_DEVICE)
+            try:
+                self.model.load_state_dict(state_dict)
+            except Exception as _:
+                state_dict = { x.replace("model.", "", 1) : v for x, v in state_dict.items() }
+                self.model.load_state_dict(state_dict)
+        ```
+        for the loading of the model to work with our fine tuned model
+        """
+        self.scorer = BARTScorer(
+            device=DEFAULT_DEVICE, checkpoint='facebook/bart-large-cnn'
+        )
+        # Path here to fine-tuend BART Model
+        self.scorer.load(
+            "/private/home/mpchen/BARTScore/train/reproduce/trained/bart_6000.pth"
```

Comment:
Ahhh we might need to upload this to the AWS bucket as well and provide a URL to it here (or otherwise download it)